src/javaharness/java/arcs/api/Schema.java

```diff
@@ -0,0 +1,79 @@
+package arcs.api;
+
+import java.util.ArrayList;
+
+public class Schema {
+    enum FieldKind {
+        PRIMITIVE, UNION, TUPLE
+    }
+    public static abstract class Field {
+        public final FieldKind kind;
+        Field(FieldKind kind) {
+            this.kind = kind;
+        }
+        public boolean isPrimitive() { return this.kind == FieldKind.PRIMITIVE; }
+        public boolean isUnion() { return this.kind == FieldKind.UNION; }
+        public boolean isTuple() { return this.kind == FieldKind.TUPLE; }
+    }
+    public static class PrimitiveField extends Field {
+        public final String type;
+        PrimitiveField(String type) {
+            super(FieldKind.PRIMITIVE);
+            this.type = type;
+        }
+    }
+    public static class UnionField extends Field {
+        public final Field types[];
```

Comment:
I would probably use List<Field> instead of primitive arrays, because you get stream operations (map, filter, etc) which you might want to use later.

Reply:
done